File: Lintcode/702-concatenated_string_with_uncommon_characters/answer.py
class Solution:
    """
    @param: : the 1st string
    @param: : the 2nd string
    @return: uncommon characters of given strings
    """

    def concatenetedString(self, s1, s2):
        # write your code here
        
        map = dict()
        result = ""
        
        for i in range(len(s1)):
            map[s1[i]] = 1
        
        for i in range(len(s2)):
            current = s2[i]
            if(current in map):
                map[current] = 0
                
        for i in range(len(s1)):
            current = s1[i]
            if(map[current] == 1):
                result += current
                
        for i in range(len(s2)):
            current = s2[i]
            if(current not in map):
                result += current
        
        
        return result

File: Lintcode/100-Remove_duplicate/answer.java
public class Solution {
    /*
     * @param nums: An ineger array
     * @return: An integer
     */
    public int removeDuplicates(int[] nums) {
        // write your code here
        int size = 0;
        
        if(nums == null || nums.length == 0)
            return size;
            
        
        for(int i = 0; i < nums.length; i++){
            if(nums[size] != nums[i]){
                nums[++size] = nums[i];
            }
        }
        
        return size+1;
    }
}